src/Microsoft.Health.Fhir.Core/Extensions/ExceptionExtensions.cs

```diff
@@ -39,5 +40,19 @@ public static bool IsRequestEntityTooLarge(this Exception e)
         {
             return e is RequestEntityTooLargeException || e?.InnerException is RequestEntityTooLargeException;
         }
+
+        public static Exception GetInnerMostException(this Exception input)
```

Comment:
>GetInnerMostException [](http://example.com/codeflow?start=32&length=21)

[https://docs.microsoft.com/en-us/dotnet/api/system.exception.getbaseexception?view=net-5.0](https://docs.microsoft.com/en-us/dotnet/api/system.exception.getbaseexception?view=net-5.0)

would it work for you?

[https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Exception.cs#L65](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Exception.cs#L65)

works a bit differently for aggregate exception, tho

[https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/AggregateException.cs#L225](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/AggregateException.cs#L225)

Reply:
System.GetBaseException() works for my purposes.